www_src/lib/dispatcher.js

```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  fire: function (eventName, eventData) {
+    var customEvent = new CustomEvent(eventName, {
+      detail: eventData
+    });
+
+    window.dispatchEvent(customEvent);
+  },
+  on: function (eventName, callback) {
+    window.addEventListener(eventName, function(e) {
+      callback.call(this, e.detail);
```

Comment:
We could have it be `window` instead. I suppose that would be marginally better. :)